### PR TITLE
Add Google's Coding Competitions presence

### PR DESCRIPTION
**Pull request by plskz**
thumbnail

# [codingcompetitions.withgoogle](https://codingcompetitions.withgoogle.com)
0 h
1 h
2 h

# Code Jam ([g.co/codejam](https://g.co/codejam))

<details>
  <summary>Home</summary>

image
</details>

<details>
  <summary>About</summary>

  image
</details>

<details>
  <summary>Schedule</summary>

  image
</details>

<details>
  <summary>FAQ</summary>

  image
</details>

<details>
  <summary>Rules</summary>

  image
</details>

<details>
  <summary>Archive</summary>

  image
  image
  image
  image
  image
</details>

<details>
  <summary>Submissions</summary>

  image
</details>

# Hash Code ([g.co/hashcode](https://g.co/hashcode))

<details>
  <summary>Home</summary>

  image
</details>

<details>
  <summary>About</summary>

  image
</details>

<details>
  <summary>Schedule</summary>

  image
</details>

<details>
  <summary>Hubs</summary>

  image
</details>

<details>
  <summary>Rules and Terms</summary>

  image
</details>

<details>
  <summary>FAQ</summary>

  image
</details>

<details>
  <summary>Archive</summary>

  image
</details>

<details>
  <summary>Scoreboard</summary>

  image
</details>

# Kick Start ([g.co/kickstart](https://g.co/kickstart))

<details>
  <summary>Home</summary>

  image
</details>

<details>
  <summary>About</summary>

  image
</details>

<details>
  <summary>Schedule</summary>

  image
</details>

<details>
  <summary>FAQ</summary>

  image
</details>

<details>
  <summary>Rules and Terms</summary>

  image
</details>

<details>
  <summary>Archive</summary>

  image
  image
  image
  image
  image
</details>

<details>
  <summary>Submissions</summary>

  image
</details>

### CODE REVIEW

**Comment by plskz** [comment]
why I see my old commits T___T

File: websites/G/Google's Coding Competitions/presence.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+          presenceData.state =
+            document.querySelector(".headline-2").textContent;
+          break;
+        }
+      }
+      break;
+    }
+
+    case "hashcode": {
+      presence = new Presence({ clientId: "848588586524868701" }); // Playing Google Hash Code
```
Comment (ririxidev) [inline websites/G/Google's Coding Competitions/presence.ts:177]: it's not good idea if you will update presence in UpdateData event :/

File: websites/G/Google's Coding Competitions/presence.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+          presenceData.state =
+            document.querySelector(".headline-2").textContent;
+          break;
+        }
+      }
+      break;
+    }
+
+    case "codejam": {
+      presence = new Presence({ clientId: "848588299147935744" }); // Playing Google Code Jam
```
Comment (ririxidev) [inline websites/G/Google's Coding Competitions/presence.ts:100]: it's not good idea if you will update presence in UpdateData event :/

File: websites/G/Google's Coding Competitions/presence.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo",
+    details: "Home page",
+    startTimestamp: timeElapsed
+  };
+
+  switch (location.pathname.split("/")[1]) {
+    case "kickstart": {
+      presence = new Presence({ clientId: "848588920437604382" }); // Playing Google Kick Start
```
Comment (ririxidev) [inline websites/G/Google's Coding Competitions/presence.ts:23]: it's not good idea if you will update presence in UpdateData event :/

**Review by rxri — CHANGES_REQUESTED** [review]

File: websites/G/Google's Coding Competitions/presence.ts (hunk 1 above)
Comment (plskz) [reply websites/G/Google's Coding Competitions/presence.ts]: sowwy

File: websites/G/Google's Coding Competitions/presence.ts [hunk 4]
```diff
@@ -5,11 +5,21 @@
  * Google Hash Code  = 848588586524868701
  * */
 
-let presence = new Presence({
-  clientId: "848587645760176148" // Playing Google's Coding Competitions
-});
+let presence: Presence;
+
+function setPresence(): void {
```
Comment (plskz) [inline websites/G/Google's Coding Competitions/presence.ts:10]: @ririxidev is this ok? xd

**Review by Bas950 — DISMISSED** [review]
LGTM

File: websites/G/Google's Coding Competitions/presence.ts (hunk 4 above)
Comment (ririxidev) [reply websites/G/Google's Coding Competitions/presence.ts:10]: yes it is

**Review by rxri — CHANGES_REQUESTED** [review]
bump version in `AtCoder` & `CodeChef`

**Review by Slowlife01 — APPROVED** [review]

**Review by rxri — APPROVED** [review]

**Pull request merged**